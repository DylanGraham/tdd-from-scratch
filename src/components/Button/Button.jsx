import * as React from "react"

const Button = () => {
  return <button className="button-primary"></button>
}

export default Button
