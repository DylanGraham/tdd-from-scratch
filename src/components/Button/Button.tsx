import * as React from "react"

interface ButtonProps {
  color?: "primary" | "secondary"
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
}

const Button: React.FunctionComponent<ButtonProps> = ({
  children,
  onClick,
  color = "primary"
}) => {
  return (
    <button
      data-testid="button"
      type="button"
      className={`button button-${color}`}
      onClick={onClick}
    >
      {children}
    </button>
  )
}

export { Button }
