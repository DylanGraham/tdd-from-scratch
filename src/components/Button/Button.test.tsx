import React from "react"
import { render, fireEvent } from "@testing-library/react"
import { Button } from "./Button"
import "@testing-library/jest-dom/extend-expect"

describe("<Button /> tests", () => {
  it("should render as default button", () => {
    // Arrange
    const content = "Hello, World!"

    // Act
    const { getByTestId, container } = render(<Button>{content}</Button>)

    // Assert
    expect(container).toMatchSnapshot()
    expect(getByTestId("button")).toHaveClass("button-primary")
    expect(container).toHaveTextContent(content)
  })

  it("should render a secondary button", () => {
    // Arrange
    const content = "Hello, World!"
    const color = "secondary"

    // Act
    const { getByTestId, container } = render(<Button color={color}>{content}</Button>)

    // Assert
    expect(container).toMatchSnapshot()
    expect(getByTestId("button")).toHaveClass(`button-${color}`)
    expect(container).toHaveTextContent(content)
  })

  it("should call the click callback handler", () => {
    // Arrange
    const content = "Hello, World!"
    const color = "secondary"
    const onClick = jest.fn()

    // Act
    const { getByTestId } = render(
      <Button color={color} onClick={onClick}>
        {content}
      </Button>
    )

    fireEvent.click(getByTestId("button"))

    // Assert
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
