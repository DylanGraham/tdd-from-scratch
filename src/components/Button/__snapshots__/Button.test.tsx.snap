// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should render a secondary button 1`] = `
<div>
  <button
    class="button button-secondary"
    data-testid="button"
    type="button"
  >
    Hello, World!
  </button>
</div>
`;

exports[`<Button /> tests should render as default button 1`] = `
<div>
  <button
    class="button button-primary"
    data-testid="button"
    type="button"
  >
    Hello, World!
  </button>
</div>
`;
